Azure/azure-cli
# [AppService] Fix #12739 az appservice list-locations returns some invalid locations

**Description**  

`az appservice list-locations` command is returning more locations than what is actually available.

Currently we are calling the /appservice/listgeoregions endpoint and returning those results. We should actually be calling that endpoint as well as the /resources/providers/get endpoint with resourceType = "sites", and returning the intersection of the two lists of regions.  

**Testing Guide**  

The list should match the list of regions options shown in the Azure Portal when you create a new App Service resource.

For example, running the command `az appservice list-locations --sku=FREE` should not be returning the location "UAE Central", since the "UAE Central" region is not an available option when creating a new App Service resource on the Portal. 

---

This checklist is used to make sure that common guidelines for a pull request are followed.
